Sources/Networking/Operations/Handling/CustomerInfoResponseHandler.swift

```diff
@@ -55,7 +70,27 @@ class CustomerInfoResponseHandler {
 
         _ = Task<Void, Never> {
             do {
-                completion(.success(try await offlineCreator.create(for: self.userID)))
+                switch result {
+                case .success:
+                    completion(.success(try await offlineCreator.create(for: self.userID)))
+                case .failure(let failure):
+                    let failIfInvalidSubscriptionKeyDetectedInDebug = self.failIfInvalidSubscriptionKeyDetectedInDebug
+
+                    if isDebug() && failIfInvalidSubscriptionKeyDetectedInDebug,
+                        case let .networkError(networkError) = failure,
+                        case let .errorResponse(errorResponse, _, _) = networkError,
+                        errorResponse.code == .invalidAppleSubscriptionKey {
+
+                        Logger.warn(
+                            Strings.configure.fail_post_receipt_due_to_invalid_in_app_purchase_key_configured_in_debug
```

Comment:
Should we log this error in non-debug builds too? not sure how often that would happen but can't hurt, right?

Reply:
I really don't think this is necessary since we're also logging [this](https://github.com/RevenueCat/purchases-ios/blob/1b0fd5c802226e403d01794671cf550613d7a6d9/Sources/Error%20Handling/ErrorCode.swift#L129) when an invalid in app purchase key is detected in both debug and non-debug builds. 

While this means we're logging two things in debug, this message clarifies that the thrown error is only thrown in debug builds. If we'd like to consolidate these messages into one message, I can easily change the original log message to say that it's an error in debug builds and remove the log message in this PR.